### ISSUE TITLE
v6.2.1: CallbackException is not invoked for unhandled exceptions from AsyncEventingBasicConsumer.Received

### ISSUE THREAD

**Issue by qrli** [rabbitmq/rabbitmq-dotnet-client#944]
rabbitmq-dotnet-client v6.2.1
rabbitmq server v3.8.3

When an unhandled exception thrown from within `AsyncEventingBasicConsumer.Received`, IModel.CallbackException will not be triggered in v6.2.1. It had always been working up to v6.1.0. 

I see you have [unit tests for this scenario for the sync BasicConsumer](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/275b4ebcc6f00df414612703ea0c3e367f233c26/projects/Unit/TestConsumerExceptions.cs), but not for the Async consumer. So the issue may be only for the async API.

**Comment by michaelklishin** [comment]
A contribution would be welcome. How unhandled exceptions are treated by async APIs in general can be very different from "regular" code.

**Comment by qrli** [comment]
I dug in the code base for sometime, searching for clues. It is my first time reading it, so still quite unfamilar.
I kind of believe I found the cause in [AsyncConsumerWorkService.cs](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/d9c6f478511f448326e02d33aa1e6f294b8a0036/projects/RabbitMQ.Client/client/impl/AsyncConsumerWorkService.cs):
```cs
                        try
                        {
                            Task task = work.Execute();
                            if (!task.IsCompleted)
                            {
                                await task.ConfigureAwait(false);
                            }
                        }
                        catch (Exception e)
```
This code tries to optimize in case Task has already finished. However, `IsCompleted` does not mean success. It can be failure with an exception. In case the handler runs fast and throws, the above logic just eats the exception inside `Task.Exception` silently.

In addition, if the task is not `await`ed, the exception will not be unwrapped, going back to the early days issues of AggregatedException parsing. 

To my knowledge, `await` has some builtin optimization in case the task has already completed. I'm not exactly sure how it compares with the manual optimization above, but from correctness point of view, I think it is easier and more robust to simply `await` the task than trying to micro-optimize here.

**Comment by michaelklishin** [comment]
Addressed in #946 contributed by @BarShavit 👏
